src/plugins/part.c

```diff
@@ -729,8 +729,10 @@ static gboolean resize_part (PedPartition *part, PedDevice *dev, PedDisk *disk,
     gint orig_flag_state = 0;
     PedSector start;
     PedSector end;
+    PedSector max_end;
     PedSector new_size = 0;
     gint status = 0;
+    const PedSector tolerance = (PedSector) (4 MiB /  dev->sector_size);
```

Comment:
It's a conservative measure, actually it would also be possible to calculate 2-3 MiB, assuming that in a logical partition the shift will be + 1 MiB offset and maybe -1 MiB end alingment, maybe 3-4 if there is already an extended partition and the caller still passes the whole disk size.